kernel/fdtable.c

```diff
@@ -374,6 +374,18 @@ int myst_fdtable_dup(
         new->device = old->device;
         new->object = newobj;
 
+        // add /proc/self/fd/[fd] entry for files
+        if (newfd != oldfd && new->type == MYST_FDTABLE_TYPE_FILE)
```

Comment:
If the oldfd is a pipe, should a link also be added?

Reply:
Adding /proc/self/fd/[fd] symlink requires the target device(the one which the fd belongs to), to support `fs_realpath`. pipdev.c does not support this currently.

Reply:
That's unfortunate. The app exposed the issue of missing /dev/fd link created a pipe, dup2 one end of it, and attempted to access it through /dev/fd/[fd].

Reply:
Is it creating a named pipe? We have an outstanding issue for that - #46.

Reply:
Not sure whether it's a named pipe. Please add an ATTN comment explaining that once support for /dev/fd/[fd] for other type of descriptors are added, the dup code here need to change.